crates/subspace-farmer/src/plotting/tests.rs

```diff
@@ -0,0 +1,343 @@
+    let plot = Plot::open_or_create(&base_directory).await.unwrap();
+    let commitments = Commitments::new(base_directory.path().join("commitments").into())
+        .await
+        .unwrap();
+    let object_mappings = ObjectMappings::open_or_create(&base_directory).unwrap();
+
+    let client = MockRpc::new();
+
+    let farmer_metadata = FarmerMetadata {
+        confirmation_depth_k: 0,
+        record_size: RECORD_SIZE as u32,
+        recorded_history_segment_size: SEGMENT_SIZE as u32,
+    };
+
+    client.send_metadata(farmer_metadata).await;
+
+    let farmer_metadata = client
+        .farmer_metadata()
+        .await
+        .expect("Could not retrieve farmer_metadata");
+
+    let identity =
+        Identity::open_or_create(&base_directory).expect("Could not open/create identity!");
+
+    let subspace_codec = SubspaceCodec::new(&identity.public_key());
+
+    let encoded_block0 = EncodedBlockWithObjectMapping {
+        block: vec![0u8; SEGMENT_SIZE / 2],
+        object_mapping: Default::default(), // This test does not concern with the object mappings at the moment.
+    };
+    let encoded_block1 = EncodedBlockWithObjectMapping {
+        block: vec![1u8; SEGMENT_SIZE / 2],
+        object_mapping: Default::default(), // This test does not concern with the object mappings at the moment.
+    };
+    let encoded_blocks = vec![encoded_block0, encoded_block1];
+
+    let new_head0 = NewHead {
+        number: "0x0".to_string(),
+    };
+    let new_head1 = NewHead {
+        number: "0x1".to_string(),
+    };
+    let new_heads = vec![new_head0, new_head1];
+
+    let plotting_instance = Plotting::start(
+        plot.clone(),
+        commitments.clone(),
+        object_mappings.clone(),
+        client.clone(),
+        farmer_metadata.clone(),
+        subspace_codec,
+    );
+
+    stream::iter(encoded_blocks)
+        .zip(stream::iter(new_heads))
+        .for_each(|(block, new_head)| {
+            let client_copy = client.clone();
+            async move {
+                client_copy.send_block(block).await;
+                client_copy.send_new_head(new_head).await;
+                // putting 500 microseconds here to give plotter some time
+                sleep(Duration::from_millis(500)).await;
+            }
+        })
+        .await;
+
+    assert_eq!(
+        plot.get_last_root_block()
+            .await
+            .unwrap()
+            .unwrap()
+            .records_root(),
+        [
+            128, 88, 79, 62, 14, 50, 76, 101, 5, 140, 34, 124, 28, 140, 2, 80, 84, 108, 192, 253,
+            210, 159, 59, 132, 116, 250, 177, 226, 192, 188, 79, 230
+        ]
+    );
+
+    // let the farmer know we are done by closing the channel(s)
+    client.drop_new_head_sender().await;
+
+    // wait for farmer to finish
+    if let Err(e) = plotting_instance.wait().await {
+        panic!("Panicked with error...{:?}", e);
+    }
+    // no need to drop plotting_instance, since the `wait` method consumes it
```

Comment:
This isn't really useful. It is impossible to drop because it is already consumed. You can also wrap part of the test with `{}` to explicitly show standalone section of the test whose internals are guaranteed to be dropped.

Reply:
I don't see the problem. It is impossible to drop, since it is already consumed. Yes? We are creating a new plotting instance anyway, what is the problem?

Reply:
The comment is useless. There is no other problem.